### PR TITLE
Make HTTP handler of embedded server to be configured

### PR DESCRIPTION
**Pull request by khos2ow**
This is to have the possibility to completely disable
HTTP and only use HTTPS.

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

## Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/apache/cloudstack/blob/master/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
Testing
- [ ] I have added tests to cover my changes.
- [ ] All relevant new and existing integration tests have passed.
- [ ] A full integration testsuite with all test that can run on my environment has passed.

### CODE REVIEW

File: client/src/main/java/org/apache/cloudstack/ServerDaemon.java [hunk 1]
```diff
@@ -129,6 +132,11 @@ public void init(final DaemonContext context) {
             setSessionTimeout(Integer.valueOf(properties.getProperty(SESSION_TIMEOUT, "30")));
         } catch (final IOException e) {
             LOG.warn("Failed to load configuration from server.properties file", e);
+        } finally {
+            // make sure that at least HTTP is enabled if both of them are set to false (misconfiguration)
+            if (!httpEnable && !httpsEnable) {
+                setHttpEnable(true);
```
[the remark this answers is not in the document]
Comment (khos2ow) [reply client/src/main/java/org/apache/cloudstack/ServerDaemon.java]: Right!

File: client/src/main/java/org/apache/cloudstack/ServerDaemon.java [hunk 2]
```diff
@@ -163,11 +171,13 @@ public void start() throws Exception {
         httpConfig.setSendDateHeader(false);
 
         // HTTP Connector
-        final ServerConnector httpConnector = new ServerConnector(server, new HttpConnectionFactory(httpConfig));
-        httpConnector.setPort(httpPort);
-        httpConnector.setHost(bindInterface);
-        httpConnector.setIdleTimeout(30000);
-        server.addConnector(httpConnector);
+        if (httpEnable) {
+            final ServerConnector httpConnector = new ServerConnector(server, new HttpConnectionFactory(httpConfig));
```
[the remark this answers is not in the document]
Comment (khos2ow) [reply client/src/main/java/org/apache/cloudstack/ServerDaemon.java]: I don't see any benefit to do so. It's an one time execution, why do you think extraction would be useful?

[the remark this answers is not in the document]
Comment (rafaelweingartner) [reply client/src/main/java/org/apache/cloudstack/ServerDaemon.java]: Well, this method has almost 100 lines. We do not have documented why some of these operations are there, and we also do not have unit tests for this part of the code (as I would expect with a big method as this one). 

Extracting blocks of code to smaller methods helps us to document and unit test the code. In this case, it is the block of code that enables and configures the HTTP connector. This single chunk can live on its own method that can be further documented and tested.

[the remark this answers is not in the document]
Comment (khos2ow) [reply client/src/main/java/org/apache/cloudstack/ServerDaemon.java]: I totally understand and agree with your point but I still don't think it applies here, these lines are basically all adjusting and configuring one logical _thing_ (embedded server) and separating them to smaller methods won't give any additional testability, documentation is a different thing which in this particular case a simple `// one liner comment` would be enough IMO.

**Comment by borisstoyanov** [comment]
@blueorangutan package

**Comment by borisstoyanov** [comment]
@blueorangutan test

File: client/src/main/java/org/apache/cloudstack/ServerDaemon.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (rafaelweingartner) [reply client/src/main/java/org/apache/cloudstack/ServerDaemon.java]: I still think that extracting code to more specific ones apply everywhere (no special cases!).  I also abhor inline comments ;)

I am not opposed to the code here, but you know, it is not a review if I am not able to share my opinion.

Thank for the PR anyways +1

**Pull request merged**